### ISSUE TITLE
improve build feedback output

### ISSUE THREAD

**Issue by stefanpenner** [testem/testem#806]
Im not 100% sure although I have some ideas on how to improve this, but let me illustrate the  issue.

In the scenario, where a build-tool is pragmatically invoking testem (and telling it to restart) the build tool may have additional feedback for the users. For example, if during test iteration the user introduces a syntax error, the build-tool will likely catch, and choose to not restart testem (since it has no valid output to run testem with again). At this point, we currently have a UX issue.

1. build tool has info it wishes to share
2. testem UI still occupies the TTY

Some work-arounds, the build-tool could at this point stop testem and display it's feedback. The downside of this, is the test-runners are also torn down. Which might be ok?

Another approach, would be for the build-tool to inform testem to display the build tools feedback as part of its own UI.

If others have more ideas here, please share.

**Comment by johanneswuerbach** [comment]
Would it be an option to render the build output as console log / error statements into the page as testem is already displaying them?

**Comment by stefanpenner** [comment]
@johanneswuerbach ya i believe that would be best. Im i don't think stoping and starting testem makes sense, as we will likely just confuse users.

Just like ember-cli, if a build error occurs, testem UI (both console, and browser) should render that information.

An idea would be, as a sibling to api.restart:

```js
api.reportError(error)
```

**Comment by johanneswuerbach** [comment]
Yep, sounds reasonable. I'll try to add those endpoints during next week as time permits :-)
